qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -0,0 +1,353 @@
+     [ 3.    0.    1.    2.    0.5 ]
+     [ 3.    2.    3.    2.    0.5 ]] 1.5
+    """
+
+    active_indices = structure.active_space(
+        mol_name,
+        hf_data,
+        n_active_electrons=n_active_electrons,
+        n_active_orbitals=n_active_orbitals,
+    )[1]
+
+    if n_active_electrons is None:
+        hf_elect_struct = MolecularData(filename=os.path.join(hf_data.strip(), mol_name.strip()))
+        n_electrons = hf_elect_struct.n_electrons
+    else:
+        n_electrons = n_active_electrons
+
+    n_spin_orbs = 2 * len(active_indices)
+    sz = np.where(np.arange(n_spin_orbs) % 2 == 0, 0.5, -0.5)
+
+    return spin2_matrix_elements(sz, n_spin_orbs), 3 / 4 * n_electrons
+
+
+def observable(me_table, init_term=0, mapping="jordan_wigner"):
+
+    r"""Builds the many-body observable whose expectation value can be
+    measured in PennyLane.
+
+    This function can be used to build second-quantized operators in the basis
+    of single-particle states (e.g., HF states) and to transform them into
+    PennyLane observables. In general, single- and two-particle operators can be
+    expanded in a truncated set of orbitals that define an active space,
+
+    .. math::
+
+        &&\hat A = \sum_{\alpha \leq 2N_\mathrm{docc}} \langle \alpha \vert \hat{\mathcal{A}}
+        \vert \alpha \rangle ~ \hat{n}_\alpha +
+        \sum_{\alpha, \beta ~ \in ~ \mathrm{active~space}} \langle \alpha \vert \hat{\mathcal{A}}
+        \vert \beta \rangle ~ \hat{c}_\alpha^\dagger\hat{c}_\beta \\
+        &&\hat B = \frac{1}{2} \left\{ \sum_{\alpha, \beta \leq 2N_\mathrm{docc}}
+        \langle \alpha, \beta \vert \hat{\mathcal{B}} \vert \beta, \alpha \rangle
+        ~ \hat{n}_\alpha \hat{n}_\beta + \sum_{\alpha, \beta, \gamma, \delta ~
+        \in ~ \mathrm{active~space}} \langle \alpha, \beta \vert \hat{\mathcal{B}}
+        \vert \gamma, \delta \rangle ~ \hat{c}_{\alpha}^\dagger \hat{c}_{\beta}^\dagger
+        \hat{c}_{\gamma} \hat{c}_{\delta} \right\}.
+
+    In the latter equations :math:`N_\mathrm{docc}` denotes the doubly-occupied orbitals,
+    if any, not included in the active space and
+    :math:`\langle \alpha \vert \hat{\mathcal{A}} \vert \beta \rangle` and
+    :math:`\langle \alpha, \beta \vert\hat{\mathcal{B}} \vert \gamma, \delta \rangle`
+    are the matrix elements of the one- and two-particle operators
+    :math:`\hat{\mathcal{A}}` and :math:`\hat{\mathcal{B}}`, respectively.
+
+    The function utilizes tools of `OpenFermion <https://github.com/quantumlib/OpenFermion>`_
+    to build the second-quantized operator and map it to basis of Pauli matrices via the
+    Jordan-Wigner or Bravyi-Kitaev transformation. Finally, the qubit operator is
+    converted to a a PennyLane observable by the function :func:`~.convert_observable`.
+
+    Args:
+        me_table (array[float]): Numpy array with the table of matrix elements.
+            For a single-particle operator this array will have shape
```

Comment:
This is really long! Is it actually helpful?

Reply:
I think it is. ``observable`` is a generic function that can be used for both, single-particle and two-particle operators. Also, if you look here https://pennylane--696.org.readthedocs.build/en/696/code/api/pennylane_qchem.qchem.obs.observable.html
it is not that bad and the info there can be very helpful to users, I think.